### PR TITLE
Sylvia/docs/student notes

### PR DESCRIPTION
**Pull request by srpearce**
@mhoeber @lamagnifica @catong @tymofij @explorerleslie @WatsonEmily 

Please review. Note that most screen shots aren't done yet and the ones that are there are just placeholders.

### CODE REVIEW

**Comment by WatsonEmily** [comment]
@jhendersonedx - please review. Thanks.

File: en_us/shared/exercises_tools/student_notes.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+.. image:: /Images/SFD_SN_bodyexample.png
+  :width: 500
+  :alt: Image of a course page that includes highlighted text and a student note
+
+.. note:: The Student Notes tool is available for text, including text in 
+ HTML components and in PDF documents. However, the tool is currently not
+ available for exercises or video transcripts.
+
+Students can access their notes either in the body of the course or on a
+specific **Notes** tab. The **Notes** tab lists all the notes that students have
```
Comment (mhoeber) [inline en_us/shared/exercises_tools/student_notes.rst]: not sure why adding "specific" notes tab.  Is there one per course? Always called "Notes"?

File: en_us/shared/exercises_tools/student_notes.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+The Student Notes tool allows students to highlight and make notes about what
+they read in the body of the course. 
+
+.. image:: /Images/SFD_SN_bodyexample.png
+  :width: 500
+  :alt: Image of a course page that includes highlighted text and a student note
+
+.. note:: The Student Notes tool is available for text, including text in 
+ HTML components and in PDF documents. However, the tool is currently not
+ available for exercises or video transcripts.
```
Comment (mhoeber) [inline en_us/shared/exercises_tools/student_notes.rst]: not available in discussions or wiki, assuming.

Comment (mhoeber) [inline en_us/shared/exercises_tools/student_notes.rst]: PDF documents -- specifically for textbooks, right? Add note to textbooks chapter?

File: en_us/students/source/SFD_student_notes.rst [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+lower-right corner of each page. By default, this option is selected, and you
+can see any notes that you make.
+
+.. image:: /Images/SFD_SN_shownotesoption.png
+  :width: 500
+  :alt: Page of course content with the pencil icon in the lower-right
+      corner
+
+If you don't want to see your highlights or notes in the course, select the
+pencil icon to turn off student notes. You can always select the icon again if
```
Comment (mhoeber) [inline en_us/students/source/SFD_student_notes.rst]: "turn off" -- I'd stick to "hide" (vs. show). the different terms confused me.

**Comment by mhoeber** [comment]
@srpearce  looks good.  couple of notes inline.  Couple other issues:
- We should add a row in the table on the course rerun chapter of course staff guide, saying whether student noes are duplicated in a rerun course (I'm assuming they are not). @lamagnifica  FYI.  I think we need to remember this table for any new feature/structure.
- Add change log for course authors (can do after merging to saudi branch -- will be easier if we all do it there (fy @catong )

**Comment by mhoeber** [comment]
@srpearce  also, after upgrading to the release that includes student notes, what happens with existing courses?  Are notes turned on for them automatically?  Will students in the middle of the course suddenly see this change?  Do we want to alert course staff to alert students to this?  Partly an issue for release notes as well.

File: en_us/shared/exercises_tools/student_notes.rst [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+The Student Notes tool allows students to highlight and make notes about what
+they read in the body of the course. 
+
+.. image:: /Images/SFD_SN_bodyexample.png
+  :width: 500
+  :alt: Image of a course page that includes highlighted text and a student note
+
+.. note:: The Student Notes tool is available for text, including text in 
+ HTML components and in PDF documents. However, the tool is currently not
```
Comment (polesye) [inline en_us/shared/exercises_tools/student_notes.rst]: We don't support PDF documents  at the moment, only `HTML components`.

File: en_us/shared/exercises_tools/student_notes.rst [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      course
+
+For detailed information about the ways students can use student notes, see
+:ref:`SFD Student Notes`.
+
+*****************************
+Enable the Student Notes Tool
+*****************************
+
+The Student Notes tool is enabled by default. You enable or disable student
```
Comment (polesye) [inline en_us/shared/exercises_tools/student_notes.rst]: `The Student Notes tool` is **[disabled](https://github.com/edx/edx-platform/blob/feature/edxnotes/common/lib/xmodule/xmodule/modulestore/inheritance.py#L178)** in course settings by default.

File: en_us/students/source/SFD_student_notes.rst (hunk 3 above)
Comment (jmclaus) [inline en_us/students/source/SFD_student_notes.rst]: @srpearce Maybe we could also tell them where the pencil icon is supposed to be situated (bottom right)?

File: en_us/students/source/SFD_student_notes.rst [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+   that appears. Your note can contain as many words as you want. Your note can
+   also contain more than one paragraph.
+
+   (screen shot)
+
+#. After you've finished writing your note, click **Save**. The Student Notes
+   editor closes, and the text that you selected remains highlighted. To see the
+   note that you created, hover over or click the highlighted text. When you
+   click the highlighted text, your note remains open until you click outside
+   the highlighted text.
```
Comment (polesye) [inline en_us/students/source/SFD_student_notes.rst]: or close the editor (Edit icon -> Cancel button).

File: en_us/students/source/SFD_student_notes.rst [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+#. When a pencil and paper "edit" icon appears above the selected text, click
+   the icon to open the student notes editor.
+
+   (screen shot)
+
+#. In the student notes editing field, type the text of your note in the field
+   that appears. Your note can contain as many words as you want. Your note can
+   also contain more than one paragraph.
+
```
Comment (jmclaus) [inline en_us/students/source/SFD_student_notes.rst]: @srpearce 

> Your note can also contain more than one paragraph.

Unfortunately, not for the moment, the default behavior of AnnotatorJS doesn't allow for multiple paragraphs. If you press ENTER, the effect will be to close the editor and save the note.

File: en_us/students/source/SFD_student_notes.rst [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+   (screen shot)
+
+#. After you've finished writing your note, click **Save**. The Student Notes
```
Comment (jmclaus) [inline en_us/students/source/SFD_student_notes.rst]: @srpearce

> After you've finished writing your note, click **Save**

or press **ENTER**

File: en_us/students/source/SFD_student_notes.rst (hunk 7 above)
Comment (jmclaus) [inline en_us/students/source/SFD_student_notes.rst]: @srpearce But SHIFT+ENTER will let you add paragraphs in the editor. Only problem, they will not be displayed afterwards.

File: en_us/shared/exercises_tools/student_notes.rst (hunk 1 above)
Comment (lamagnifica) [inline en_us/shared/exercises_tools/student_notes.rst]: "all the notes that students have made" -- does that mean, each student's notes are public, part of a massive course-wide record? From the demo, I thought that each student would be able to see only their own notes on the Notes tab.
Also, is there a way that course teams can get access to the notes?

File: en_us/shared/exercises_tools/student_notes.rst [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  :width: 500
+  :alt: Image of a course page that includes highlighted text and a student note
+
+.. note:: The Student Notes tool is available for text, including text in 
+ HTML components and in PDF documents. However, the tool is currently not
+ available for exercises or video transcripts.
+
+Students can access their notes either in the body of the course or on a
+specific **Notes** tab. The **Notes** tab lists all the notes that students have
+made during the course. It also contains links to the notes in the body of the
```
Comment (lamagnifica) [inline en_us/shared/exercises_tools/student_notes.rst]: what is the difference between these two types of notes? could the final sentence be about the highlighting as distinct from the notes?
ah, looking at the image, I'd suggest something like "It also provides a link to the location of each note in the course content."

File: en_us/shared/exercises_tools/student_notes.rst (hunk 5 above)
Comment (lamagnifica) [inline en_us/shared/exercises_tools/student_notes.rst]: You enable or disable ^the student...

**Comment by lamagnifica** [comment]
StNotesIcon.png doesn't seem necessary

File: en_us/shared/exercises_tools/create_exercises_and_tools.rst [hunk 10]
```diff
@@ -76,6 +76,12 @@ General Exercises and Tools
           :alt: Example problem written in LaTeX
      - :ref:`Problem Written in LaTeX`
      - If you have an problem that is already written in LaTeX, you can use this problem type to easily convert your code into XML.
+   * - .. image:: /Images/StudentNotes.png
+          :width: 100
+          :alt: Image of student notes in a course
+     - :ref:`Student Notes Tool`
+     - With student notes, students can highlight and make notes about what they
```
Comment (lamagnifica) [inline en_us/shared/exercises_tools/create_exercises_and_tools.rst]: The current construction is a bit awkward, maybe because "and make notes about" seems secondary to highlighting:
"With student notes, students can highlight, and make notes about, what they read.."
I think I would invert the order of these two features (throughout) to something more like, 
...students can make notes about what they read and highlight passages in the course content.

File: en_us/students/source/SFD_student_notes.rst [hunk 11]
```diff
@@ -0,0 +1,130 @@
+.. _SFD Student Notes:
+
+##############################
+Student Notes
+##############################
+
+Your edX course may include student notes, a feature that you can use to
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: I'd suggest rewording this introductory statement to make it simpler:
Your edX course may include student notes, a feature that you can use to  --> In some edX courses, you can

File: en_us/students/source/SFD_student_notes.rst [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+highlight and make notes about what you read during your course. Student
+notes allow you to record your thoughts about course content right in the body
+of the course. If you leave a page and come back to it later, you can still see
+your note on the page.
+
+.. image:: /Images/SFD_SN_bodyexample.png
+  :width: 500
+  :alt: Image of a course page that includes highlighted text and a student note
+
+You can also access a tab that lists all the notes you've made during the
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: I'd explicitly give the name of the tab. Also, careful of the contractions.

File: en_us/students/source/SFD_student_notes.rst [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+.. note:: You can create notes for most text in the body of the course. 
+ However, the student notes feature is currently not available for exercises or
+ videos.
+
+
+***************************
+Using Student Notes
+***************************
+
+===================
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: I'd suggest reorganizing somewhat:
- how to tell whether the course offers this feature (a pencil icon displays) -- move lines 38-40 up under the Using Student Notes heading as an intro statement, 'cause if the course doesn't offer the feature, there's no need to read any of the rest of this section
- how to make a note or highlight a passage -- I think they are likely to want to try it out before deciding to hide it...
- how to view -- now that I've made a note, where do I see it
- how to edit or delete -- less frequently needed than the above
- how to hide the feature -- since it's on by default, put the emphasis on hiding it, but include how to show it again

File: en_us/students/source/SFD_student_notes.rst [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+.. image:: /Images/SFD_SN_shownotesoption.png
+  :width: 500
+  :alt: Page of course content with the pencil icon in the lower-right
+      corner
+
+If you don't want to see your highlights or notes in the course, select the
+pencil icon to turn off student notes. You can always select the icon again if
+you want to see your notes.
+
+.. note:: If you turn off student notes, you cannot see any of the notes
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: I don't think this note is really necessary. If it is, wouldn't it be better placed as an introduction to the concept of what it means to hide your notes?

File: en_us/students/source/SFD_student_notes.rst [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+If you don't want to see your highlights or notes in the course, select the
+pencil icon to turn off student notes. You can always select the icon again if
+you want to see your notes.
+
+.. note:: If you turn off student notes, you cannot see any of the notes 
+ that you have made in the course. You also cannot create new notes while the
+ feature is turned off.
+
+===================
+Create a Note
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: Make a Note? Write a Note?
"make" seems more the way one would think about the action or task than "create"

File: en_us/students/source/SFD_student_notes.rst [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+#. Select the text that you want to create the note about. You can select as
+   many characters and words as you want.
+
+#. When a pencil and paper "edit" icon appears above the selected text, click
+   the icon to open the student notes editor.
+
+   (screen shot)
+
+#. In the student notes editing field, type the text of your note in the field
+   that appears. Your note can contain as many words as you want. Your note can
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: In the student notes editing field, type the text of your note in the field that appears. --> Type your note into the field that appears.
..and, @jmclaus , there's really no maximum limit on character count?

File: en_us/students/source/SFD_student_notes.rst (hunk 6 above)
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: oh. hunh.
I thought the highlighting and the note taking were separate things that you could accomplish
If you highlight text, but do not make a note, does the selected text remain highlighted, appear on the Notes tab, have a link, etc.? or is the highlighting lost?

File: en_us/students/source/SFD_student_notes.rst [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+
+#. After you've finished writing your note, click **Save**. The Student Notes
+   editor closes, and the text that you selected remains highlighted. To see the
+   note that you created, hover over or click the highlighted text. When you
+   click the highlighted text, your note remains open until you click outside
+   the highlighted text.
+
+=======================
+Edit or Delete a Note
+=======================
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: re: the, can you just highlight something question
might need to explicitly say how you get rid of highlighting 
maybe separate sections for 
Edit a Note
Delete a Note or Highlighting for a Passage

File: en_us/students/source/SFD_student_notes.rst [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+
+To edit a note that you've created:
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: strike "that you've created"

File: en_us/students/source/SFD_student_notes.rst [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+
+#. Hover over the highlighted text until your note appears.
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: 1. Use the Notes tab or otherwise locate the course content where your note appears...

File: en_us/students/source/SFD_student_notes.rst [hunk 20]
```diff
@@ -0,0 +1,130 @@
...
+#. When the note appears, hover over the note until a pencil icon and an X icon
+   appear in the note field, and then click the pencil icon to open the note
+   editor.
+
+   (screen shot)
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst]: If this is really similar-looking to the add process, no need for an additional image

File: en_us/students/source/SFD_student_notes.rst [hunk 21]
```diff
@@ -0,0 +1,130 @@
...
+
+#. In the note editor, edit your note, and then click **Save**.
+
+To delete a note:
+
+#. Hover over the highlighted text until your note appears.
+#. When the note appears, hover over the note until a pencil icon and an X icon
+   appear in the note field, and then click the X icon to delete the note.
+
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst:72]: a double hover? how would this work for someone who can't use a mouse?

File: en_us/students/source/SFD_student_notes.rst [hunk 22]
```diff
@@ -0,0 +1,130 @@
...
+===========
+View Notes
+===========
+
+You can view the notes you've added to your course in two places. 
+
```
Comment (lamagnifica) [inline en_us/students/source/SFD_student_notes.rst:80]: ...the notes you've added to your course... --> ...your notes...
careful of enumerating how many items are in a list, as they change 
Actually, I'd suggest deemphasizing that there are two different ways to see note. Being able to see them where you make them in the course content is kind of an obvious thing -- worth stating, but not calling out as on a par with the navigate-to-separately Notes tab. 
Maybe replace lines 106-114 with, "If you can make notes in your course, a Notes tab appears at the top of every page. On the Notes tab..."
then on line 129, say something to the effect of, As you read course content, any notes that you previously made appear.
?

**Comment by lamagnifica** [comment]
Suggestions throughout.

**Comment by jhendersonedx** [comment]
:+1: 

Very clear layout for these tools. Summaries work well.

File: en_us/students/source/SFD_student_notes.rst (hunk 7 above)
Comment (srpearce) [inline en_us/students/source/SFD_student_notes.rst]: @jmclaus OK - I didn't realize we hadn't changed that (I remember filing the ticket requesting the change but hadn't been following it). I just deleted the part about multiple paragraphs since I'd rather not mention it than have to say "yeah, this doesn't quite work the way you might expect..."

File: en_us/students/source/SFD_student_notes.rst (hunk 6 above)
Comment (srpearce) [inline en_us/students/source/SFD_student_notes.rst]: A student can just enter a single space and close the student notes editor (I added this info).

File: en_us/students/source/SFD_student_notes.rst (hunk 8 above)
Comment (polesye) [inline en_us/students/source/SFD_student_notes.rst]: > or press ENTER

We will change it to `Ctrl + ENTER` or `Cmd + ENTER` in the next sprint. See 
https://openedx.atlassian.net/browse/TNL-1078

File: en_us/students/source/SFD_student_notes.rst (hunk 7 above)
Comment (polesye) [inline en_us/students/source/SFD_student_notes.rst]: We expect to fix it in the next sprint.

File: en_us/shared/exercises_tools/student_notes.rst (hunk 4 above)
Comment (explorerleslie) [inline en_us/shared/exercises_tools/student_notes.rst]: +1

File: en_us/shared/exercises_tools/student_notes.rst (hunk 2 above)
Comment (explorerleslie) [inline en_us/shared/exercises_tools/student_notes.rst]: You cannot make notes in PDF documents, including textbooks, at this time.

File: en_us/shared/exercises_tools/student_notes.rst (hunk 1 above)
Comment (explorerleslie) [inline en_us/shared/exercises_tools/student_notes.rst]: Would prefer it to be just "the Notes tab".

Comment (explorerleslie) [inline en_us/shared/exercises_tools/student_notes.rst]: Agree that with @lamagnifica that the phrase "all the notes that students have made during the course" is misleading -- this is just the notes that an individual student makes in the courseware, so they can see their notes. No sharing of notes.

File: en_us/students/source/SFD_student_notes.rst (hunk 14 above)
Comment (explorerleslie) [inline en_us/students/source/SFD_student_notes.rst]: agree that this probably isn't needed.

File: en_us/students/source/SFD_student_notes.rst (hunk 7 above)
Comment (explorerleslie) [inline en_us/students/source/SFD_student_notes.rst]: It's a current bug. I say we update the documentation with "also contain more than one paragraph" after the bug is fixed.

File: en_us/students/source/SFD_student_notes.rst (hunk 6 above)
Comment (explorerleslie) [inline en_us/students/source/SFD_student_notes.rst]: @lamagnifica the former is correct: "the selected text remain highlighted, appear on the Notes tab, have a link, etc."

File: en_us/students/source/SFD_student_notes.rst (hunk 17 above)
Comment (explorerleslie) [inline en_us/students/source/SFD_student_notes.rst]: Nope, no need to have a different action. If there's no note, it will just say "No comment" in the note area and you can edit or delete the highlight from there, same as you would a note.

File: en_us/students/source/SFD_student_notes.rst (hunk 21 above)
Comment (explorerleslie) [inline en_us/students/source/SFD_student_notes.rst:72]: You can click the note to have it remain open until you click outside the note as well as hovering, avoiding the double-hover.

File: en_us/students/source/SFD_student_notes.rst (hunk 13 above)
Comment (explorerleslie) [inline en_us/students/source/SFD_student_notes.rst]: I totally agree with @lamagnifica 's comments here. This is currently structured in a sort of funny way compared to the way that students will actually encounter and use the feature, so the workflow above matches my imagining of the student's mental model better.

**Comment by explorerleslie** [comment]
@srpearce looking good! Made a bunch of comments, will wait for a re-ping for a re-review.

File: en_us/students/source/SFD_student_notes.rst (hunk 7 above)
Comment (srpearce) [inline en_us/students/source/SFD_student_notes.rst]: +1 to @explorerleslie's  comment - that's the current plan.

**Comment by srpearce** [comment]
@lamagnifica @explorerleslie Ready for final review (including screen shots). Tried to build a draft HTML file so you could see it, but RTD isn't cooperating well.

File: en_us/shared/exercises_tools/student_notes.rst [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+The Student Notes tool allows students to highlight and make notes about what
+they read in the body of the course. 
+
+.. image:: /Images/SFD_SN_bodyexample.png
+  :width: 500
+  :alt: Image of a course page that includes highlighted text and a student note
+
+.. note:: The Student Notes tool is available for text, including text in 
+ HTML components. However, the tool is currently not available for discussions,
+ exercises, or video transcripts.
```
Comment (explorerleslie) [inline en_us/shared/exercises_tools/student_notes.rst]: YOu may want to explicitly say that student notes isn't available in pdf textbooks. It might be a common question.

File: en_us/students/source/SFD_student_notes.rst [hunk 24]
```diff
@@ -0,0 +1,142 @@
...
+highlight and make notes about what you read during your course. Student
+notes allow you to record your thoughts about course content right in the body
+of the course. If you leave a page and come back to it later, you can still see
+your note on the page.
+
+.. image:: /Images/SFD_SN_bodyexample.png
+  :width: 500
+  :alt: Image of a course page that includes highlighted text and a student note
+
+.. note:: You can create notes for most text in the body of the course.
```
Comment (explorerleslie) [inline en_us/students/source/SFD_student_notes.rst:16]: same note about not being able to use student notes in pdf textbooks, above.

File: en_us/students/source/SFD_student_notes.rst [hunk 25]
```diff
@@ -0,0 +1,142 @@
...
+
+##############################
+Student Notes
+##############################
+
+In some edX courses, you can use the Student Notes feature to
+highlight and make notes about what you read during your course. Student
+notes allow you to record your thoughts about course content right in the body
+of the course. If you leave a page and come back to it later, you can still see
+your note on the page.
```
Comment (explorerleslie) [inline en_us/students/source/SFD_student_notes.rst]: Is this last sentence necessary? Seems like a pretty useless feature if all your notes disappear after you've made them. ;)

**Comment by explorerleslie** [comment]
two minor nits, otherwise :+1: . Great work @srpearce !

File: en_us/shared/exercises_tools/student_notes.rst (hunk 23 above)
Comment (srpearce) [inline en_us/shared/exercises_tools/student_notes.rst]: OK, done.

File: en_us/students/source/SFD_student_notes.rst (hunk 24 above)
Comment (srpearce) [inline en_us/students/source/SFD_student_notes.rst:16]: Done.

File: en_us/students/source/SFD_student_notes.rst (hunk 25 above)
Comment (srpearce) [inline en_us/students/source/SFD_student_notes.rst]: (sheepishly) Yeah... I think I was trying to say "you can see what you were thinking when you read this content earlier." Lemme see if I can think of a better way to communicate that, or say something else instead.
